### PR TITLE
Remove Tox environments now handled by precommit and CI config

### PR DESCRIPTION
**Pull request by wsanchez**
Drop some ballast in Tox file.

This also expends the test matrix for recent python versions to all supported Twisted versions, and adds pypy-3.9.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -209,15 +139,15 @@ commands =
...
+    coverage-py{37,38,39,310,311,py3}-tw{1,2}{0,1,2,3,4,5,6,7,8,9}{0,1,2,3,4,5,6,7,8,9}
+    coverage-py{37,38,39,310,311,py3}-tw{current,trunk}
 
 basepython = {[default]basepython}
 
 skip_install = True
 
 deps =
-    coverage
+    -r requirements/coverage.txt
```
Comment (adiroiban) [inline tox.ini:150]: Very minor comment.
I think that this if fine, and we can merge it as it is.

My preference, is to define an `extras = coverage` or just a simple `extras = dev` and in setup.py define all the development/testing  related packages as an `extra_require`.
In this way, all the deps are available inside the same file... and it should be easier to review and discover the deps.

**Review by adiroiban — APPROVED** [review]
Thanks for the cleanup.

It looks good and I think it can be merged.

File: tox.ini (hunk 1 above)
Comment (wsanchez) [reply tox.ini:150]: That's inconsistent with the other tox environments.

I don't think need we need to expose tox test environments in setup.py as extras, so I would hesitant to move that change.

Comment (adiroiban) [reply tox.ini:150]: Those dependencies are supposed to be used for local test run and CI test runs.

When doing local dev, I also want to run the coverage reports locally.

**Pull request merged**